src/main/java/baritone/pathing/movement/MovementHelper.java

```diff
@@ -344,6 +347,25 @@ static boolean canWalkOn(BlockStateInterface bsi, int x, int y, int z, IBlockSta
         return block instanceof BlockStairs;
     }
 
+    static List<Item> filterThrowawayForPlacing(List<Item> items) {
```

Comment:
Maybe this should be split into `filterThrowawayForPlacing` and an overload of `canWalkOn` (taking only a blockstate) so if we need to check single blocks we don't face the same problem once again.